### DISCUSSION TITLE
Accuracy issue with Milvus 2.0.2 compare to FAISS 1.6.5

### DISCUSSION THREAD

**Discussion by dilip814** [milvus-io/milvus#18241] [Q&A and General discussion]
Hello, 

we are planning to migrate from FAISS to Milvus and observed that distance reported by Milvus similarity search is differ w.r.t FAISS and we see higher number of false positive with Milvus compared to FAISS and also similarity_cos distance also have variance.

False positive
Current file:  amit7.raw
Given embeddings' current_name: Amit, currrent_chance: [[0.6700583  0.6200403  0.58369577]] using FAISS
Faiss_name Amit Faiss_chance [[0.6700583  0.6200403  0.58369577]]
Milvus name and chances ([[434283938726605730, 434283938726605745, 434375500312019979]], [[0.6701831221580505, 0.6200402975082397, 0.5972496271133423]], [{'FirstName': 'Amit', 'LastName': 'Rozner', 'PersonKey': '121f496a-f915-11ec-a5c9-d2a5032b8e9c', 'Categories': {'Name': 'Employee', 'Value': '[129,59,59]'}, 'Enabled': True}])

Similarity_cos difference:
Current file:  kash15.raw
Given embeddings' current_name: Kash, currrent_chance: [[0.6561022  0.6064243  0.60522723]] using FAISS
Faiss_name Kash Faiss_chance [[0.6561022  0.6064243  0.60522723]]
Milvus name and chances ([[434283938726616322, 434283938726616666, 434283938726616583]], [[0.35849156975746155, 0.30538883805274963, 0.3027399182319641]], [{'FirstName': 'Kash', 'LastName': 'T', 'PersonKey': '119b6374-f916-11ec-a95c-d2a5032b8e9c', 'Categories': {'Name': 'Manager', 'Value': '[30,207,230]'}, 'Enabled': True}])

**Comment by yhmo** [comment]
Could you provide your testing script?
From your description, I couldn't know how many embeddings in database. Range of Euclidean distance should be 0~+∞, 0 is the most similar, but you get result [[0.6701831221580505, 0.6200402975082397, 0.5972496271133423]], the three distance values are arranged from large to small.
The search parameter "topk" is 3?
